maskrcnn_benchmark/layers/batch_norm.py

```diff
@@ -17,6 +17,13 @@ def __init__(self, n):
         self.register_buffer("running_var", torch.ones(n))
 
     def forward(self, x):
+        # Cast all fixed parameters to half() if necessary
+        if x.type() == torch.half:
+            self.weight = self.weight.half()
```

Comment:
So it seems that we don't explicitly cast the model to `fp16` during initialization, is that right?
This seems a bit counter-intuitive to me, what happens if we just cast everything in `model` to `.half()`?

Reply:
In using "support" I chose my words badly. It's not necessarily support, more "can be used with a reasonable expectation of not losing accuracy". Apex.amp takes a conservative approach by not moving ops to fp16 when we're not sure of their accuracy (the lists of what is / isn't moved to fp16 are in the files [here](https://github.com/NVIDIA/apex/tree/master/apex/amp/lists).

We could try casting the entire model to half and see what happens -- there's enough code in the RPN especially that I'm just not sure how it'll behave in lower precisions that I decided to be conservative beyond what apex.amp does. Unfortunately that means until PyT can grok `y_16 = a_32 * x_16 + b_32` where the subscripts denote precision we have to do something manual here